### DISCUSSION TITLE
Using matadata for file created by another app

### DISCUSSION THREAD

**Discussion by Archie-2021** [SilentVoid13/Templater#646] [Help] [answered]
I am having a big problem with making use of Templater with other apps, I can't access file metadata like filename or yaml from it because they are not there at the moment for Templater to get them. Like I am trying to use it with daily planner and using the title of the file (which is a date) in my Templater scripts down below. How can I time it correctly so that they get executed after the other stuff like title are already there, I tried sleep and await but couldn't make it work. 
Please help

**Comment by dredhorse** [comment, marked as answer]
for the title take a look at https://github.com/SilentVoid13/Templater/discussions/632#discussion-4060197
for the rest... I haven't found a solution myself. 
Looks like Obsidian uses a cached version of the note for most of the plugin related calls, so a lot of plugins are failing to read the YAML or filename when you just created it.

**Reply by Archie-2021** [reply]
> for the title take a look at [#632 (comment)](https://github.com/SilentVoid13/Templater/discussions/632#discussion-4060197) for the rest... I haven't found a solution myself. Looks like Obsidian uses a cached version of the note for most of the plugin related calls, so a lot of plugins are failing to read the YAML or filename when you just created it.

It works , thanks. For metadata I can make a user script to fetch it. I already have one for QuickAdd. It is such a shame that Templater native functions can't be used...

**Comment by dredhorse** [comment]
What do you mean by this? The Script looks good, it contains probably some needed fixes for templater.

You can probably try: const myOutput = myYaml.frontmatter[myName];

I can’t find this post on Github anymore.

**Reply by Archie-2021** [reply]
> What do you mean by this? The Script looks good, it contains probably some needed fixes for templater. You can probably try: const myOutput = myYaml.frontmatter[myName]; I can’t find this post on Github anymore.
> […](#)
> On 26. May 2022, 19:24 +0200, Archie-2021 ***@***.***>, wrote: > > for the title take a look at #632 (comment) for the rest... I haven't found a solution myself. Looks like Obsidian uses a cached version of the note for most of the plugin related calls, so a lot of plugins are failing to read the YAML or filename when you just created it. > It works , thanks. For metadata I can make a user script to fetch it. I already have one for QuickAdd. It is such a shame that Templater native functions can't be used... i made this script to get the frontmatter, the issue is that the command i know of which fetches the one from the list uses literal name of the front matter and i don't know how to use a variable input to get it. function yamlFetch(myName){ const activeFile = app.workspace.getActiveFile(); const myYaml = app.metadataCache.getFileCache(activeFile); const myOutput = myYaml.frontmatter.myName; return myOutput; } module.exports = yamlFetch; — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

Yes I forgot about that. It is working now, thanks
